common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -300,6 +302,16 @@ protected void renewAccessToken(@NonNull final SilentTokenCommandParameters para
 
             // Set the AuthenticationResult on the final result object
             acquireTokenSilentResult.setLocalAuthenticationResult(authenticationResult);
+        }else {
+            if (tokenResult.getErrorResponse() != null) {
+                final String errorCode = tokenResult.getErrorResponse().getError();
+                final String subErrorCode = tokenResult.getErrorResponse().getSubError();
+
+                if (errorCode.equals(INVALID_GRANT) && subErrorCode.equals(BAD_TOKEN)) {
+                    Logger.info(TAG, "Refresh token is invalid, deleting from cache");
+                    tokenCache.removeCredential(cacheRecord.getRefreshToken());
+                }
```

Comment:
Can we log the contents of the `sub_error`?

Reply:
logging both error and suberror now